### PR TITLE
complete Feat/qa

### PR DESCRIPTION
**Pull request by yulanrong**
@ericjlie @vmichaelwu Hi I finished all the components in qa module. It's a large chunk of code changed sorry! Can you take a look at them? many thanks!

### CODE REVIEW

File: client/src/component/qa/AddAnswer.jsx [hunk 1]
```diff
@@ -1,41 +1,122 @@
...
 import axios from 'axios';
 
 const Container = styled.div`
-  display: flex;
-  justify-content: center;
-  align-items: center;
-  height: 100vh;
+position: fixed;
+z-index: 100;
+top: 0;
```
Comment (ericjlie) [inline client/src/component/qa/AddAnswer.jsx]: these are fine but need an indentation for the css ie.
const Container = styled.div`
    position: fixed;
    z-index: 100;
    top: 0;
`

Comment (ericjlie) [reply client/src/component/qa/AddAnswer.jsx:8]: oops that didn't render correctly, but just add a tab before each line

File: client/src/component/qa/AddAnswer.jsx [hunk 2]
```diff
@@ -1,41 +1,122 @@
...
   const postAnswer = (() => {
-    axios.post().then().catch();
-    handleAddAnswer();
+    const newAnswer = {
+      body: answer,
+      name,
+      email,
+    };
+    axios.post(`http://localhost:3000/qa/questions/answers?question_id=${questionId}`, newAnswer).then(() => {
+      console.log('sent');
```
Comment (ericjlie) [inline client/src/component/qa/AddAnswer.jsx]: Remove console log

File: client/src/component/qa/AddQuestion.jsx [hunk 3]
```diff
@@ -45,57 +70,62 @@ const AddQuestion = ({ productId }) => {
...
-      product_id: productId,
+      product_id: Number(productId),
     };
 
-    if (question.length < 1 || name.length < 1 || email.length < 1 || !validateEmail(email)) {
-      return <p>You must enter the following:</p>;
-    }
-    axios.post('/qa/questions', newQuestion).then(() => {
+    axios.post('http://localhost:3000/qa/questions', newQuestion).then(() => {
       console.log('sent');
```
Comment (ericjlie) [inline client/src/component/qa/AddQuestion.jsx]: Another console log

File: client/src/component/qa/AddQuestion.jsx [hunk 4]
```diff
@@ -45,57 +70,62 @@ const AddQuestion = ({ productId }) => {
...
+                <input style={{ width: '80%', height: '8%' }} maxLength="60" placeholder="Example@example.com" name="email" required onChange={(e) => { handleAddQuestion(e); }} />
+                <p style={{ fontSize: '10px' }}>For authentication reasons, you will not be emailed.</p>
+                <button
+                  style={{ fontSize: '15px', cursor: 'pointer' }}
+                  type="submit"
+                >
+                  Submit
+
+                </button>
+                <button style={{ marginLeft: '50px', fontSize: '15px', cursor: 'pointer' }} type="button" onClick={handleClose}>Cancel</button>
```
Comment (ericjlie) [inline client/src/component/qa/AddQuestion.jsx:122]: i think it's fine to merge these anyways, but if you have a lot of extra time on your hands, inline styling is not great and should be converted to styled components at least.
https://www.quora.com/Is-it-really-a-bad-practice-to-use-inline-styles-in-HTML

**Review by ericjlie — COMMENTED** [review]
@yulanrong finished reviewing. Functionally looks good to me, though two formatting things I see are:
- place indentations before the css in a styled component definition
- if you have extra time refactor inline styling to styled components, but if not the performance gain isn't worth all the effort.

And also a couple of console logs that should be removed.

File: client/src/component/qa/AddQuestion.jsx (hunk 4 above)
Comment (yulanrong) [reply client/src/component/qa/AddQuestion.jsx:122]: Will do. thanks

**Pull request merged**